solutions/cses-2428.mdx

```diff
@@ -0,0 +1,77 @@
+---
+id: cses-2428
+source: CSES
+title: Subarray Distinct Values
+author: Sofia Yang
+---
+
+## Explanation
+
+Use a sliding window and keep a multiset to record the frequency of each element in the array.
+Expand the window to the right if the number of distinct elements after expanding the window is less or equal to $K$,
+otherwise remove the left-most element.
+When a new element at index $i$ is added to a window with a left-most element of index $j$,
+it will create ($i$ - $j$) new subarrays, i.e ($j$ to $i$), ($j + 1$ to $i$), etc. which we can use to calculate the number of valid subarrays.
```

Comment:
```suggestion
it will create ($i - j$) new subarrays, i.e ($j$ to $i$), ($j + 1$ to $i$), etc. which we can use to calculate the number of valid subarrays.
```